identity_stardust/src/did/stardust_did.rs

```diff
@@ -0,0 +1,903 @@
+  pub fn new(bytes: &[u8; 32], network_name: &NetworkName) -> Self {
+    let tag = prefix_hex::encode(bytes);
+    let did: String = format!("did:{}:{}:{}", Self::METHOD, network_name, tag);
+
+    Self::parse(did).expect("DIDs constructed with new should be valid")
+  }
+
+  /// Parses an [`StardustDID`] from the given `input`.
+  ///
+  /// # Errors
+  ///
+  /// Returns `Err` if the input does not conform to the [`StardustDID`] specification.
+  pub fn parse(input: impl AsRef<str>) -> Result<Self> {
+    CoreDID::parse(input).and_then(Self::try_from_core)
+  }
+
+  /// Creates a new placeholder [`StardustDID`] with the given network name.
+  ///
+  /// # Example
+  ///
+  /// ```
+  /// # use identity_did::did::DID;
+  /// # use identity_stardust::NetworkName;
+  /// # use identity_stardust::StardustDID;
+  /// #
+  /// let placeholder = StardustDID::placeholder(&NetworkName::try_from("smr").unwrap());
+  /// assert_eq!(placeholder.as_str(), "did:stardust:smr:0x0000000000000000000000000000000000000000000000000000000000000000");
+  pub fn placeholder(network_name: &NetworkName) -> Self {
+    Self::new(&[0; 32], network_name)
+  }
+
+  // Check if the tag matches a potential alias_id
+  fn check_tag_str(tag: &str) -> Result<()> {
+    prefix_hex::decode::<[u8; TAG_BYTES_LEN]>(tag)
+      .map_err(|_| DIDError::InvalidMethodId)
+      .map(|_| ())
+  }
+
+  // Normalizes the DID `method_id` by removing the default network segment if present.
+  //
+  // E.g.
+  // - `"did:stardust:main:123" -> "did:stardust:123"` is normalized
+  // - `"did:stardust:dev:123" -> "did:stardust:dev:123"` is unchanged
+  // TODO: Remove the lint once this bug in clippy has been fixed. Without to_owned a mutable reference will be aliased.
+  #[allow(clippy::unnecessary_to_owned)]
+  fn normalize(mut did: CoreDID) -> CoreDID {
+    let method_id = did.method_id();
+    let (network, tag) = Self::denormalized_components(method_id);
+    if tag.len() == method_id.len() || network != NetworkName::DEFAULT_STR {
+      did
+    } else {
+      did
+        .set_method_id(tag.to_owned())
+        .expect("normalizing a valid CoreDID should be Ok");
+      did
+    }
+  }
+
+  /// Checks if the given `DID` has a valid [`StardustDID`] network name.
+  ///
+  /// # Errors
+  ///
+  /// Returns `Err` if the input is not a valid network name according to the [`StardustDID`] method specification.
+  pub fn check_network<D: DID>(did: &D) -> Result<()> {
+    let network_name = Self::denormalized_components(did.method_id()).0;
+    NetworkName::validate_network_name(network_name).map_err(|_| DIDError::Other("invalid network name"))
+  }
+
+  /// Checks if the given `DID` is syntactically valid according to the [`StardustDID`] method specification.
+  ///
+  ///
+  ///  This function does NOT check whether the `did-method-specific-id` corresponds to a Blake2b-256 hash of an
+  ///  `OutputID`, despite this being a requirement for the [`StardustDID`] method specification.
```

Comment:
This is misleading, the `check_tag_str` function does ensure the tag is well-formed (i.e. 'syntactically valid'), just not whether it is an existing `AliasId`.

```suggestion
```

Reply:
Should I reword this to `This function does NOT check whether the `did-method-specific-id` corresponds to an existing `AliasId` or would it be better to simply remove this text altogether?